Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -7648,16 +7648,27 @@ class MacroAssemblerX86_64 : public AbstractMacroAssembler<Assembler> {
 
     void add64(FPRegisterID left, FPRegisterID right, FPRegisterID dest)
     {
-        UNUSED_PARAM(left);
-        UNUSED_PARAM(right);
-        UNUSED_PARAM(dest);
+        if (supportsAVX()) {
+            m_assembler.vpaddq_rrr(right, left, dest);
+            return;
+        }
+
+        if (left == dest && right == dest)
+            m_assembler.paddq_rr(dest, dest);
+        else if (left == dest)
+            m_assembler.paddq_rr(right, dest);
+        else if (right == dest)
+            m_assembler.paddq_rr(left, dest);
+        else {
+            m_assembler.movaps_rr(left, dest);
+            m_assembler.paddq_rr(right, dest);
+        }
     }
 
     void sub64(FPRegisterID left, FPRegisterID right, FPRegisterID dest)
     {
-        UNUSED_PARAM(left);
-        UNUSED_PARAM(right);
-        UNUSED_PARAM(dest);
+        ASSERT(supportsAVX());
```

Comment:
How do we know that `sub64` is only used when AVX is available?  What do I need to grep / look at to confirm this?

Also, don't we need to update AirOpcode.opcodes?  Currently, it says:
```
arm64: Sub64 U:G:64, U:G:64, D:G:64
    Tmp, Tmp, Tmp
    Tmp, Imm, Tmp
```